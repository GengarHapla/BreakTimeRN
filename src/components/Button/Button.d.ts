import { ReactChild } from 'react';

export interface IButtonProps {
    onPress: any;
    style?: any;
    disabled?: boolean;
    children: ReactChild;
}
