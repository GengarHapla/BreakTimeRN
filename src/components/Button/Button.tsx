import * as React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { IButtonProps } from './Button.d';

export const Button = ({ children, onPress, style, disabled }: IButtonProps) => {
    return (
        <TouchableOpacity
            onPress={onPress}
            style={[styles.button, ...style]}
            disabled={disabled}
        >
            {children}
        </TouchableOpacity>
    )
};

const styles = StyleSheet.create({
    button: {
        padding: 6,
        borderRadius: 4,
        opacity: 0.8,
        alignItems: 'center',
    }
});
