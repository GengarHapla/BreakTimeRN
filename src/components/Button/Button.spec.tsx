import React from 'react';
import { Button } from './Button';

import renderer from 'react-test-renderer';
import { View } from 'react-native';

test('renders correctly', () => {
    const tree = renderer.create(<Button onPress={() => {}}><View/></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});
