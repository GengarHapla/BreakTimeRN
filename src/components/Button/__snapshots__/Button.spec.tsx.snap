// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 4,
      "opacity": 0.8,
      "padding": 6,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View />
</View>
`;
